sound/soc/soc-pcm.c

```diff
@@ -1571,8 +1571,11 @@ void dpcm_be_dai_stop(struct snd_soc_pcm_runtime *fe, int stream,
 			}
 		}
 
-		__soc_pcm_close(be, be_substream);
-		be_substream->runtime = NULL;
+		if (be_substream) {
+			__soc_pcm_close(be, be_substream);
+			be_substream->runtime = NULL;
```

Comment:
Line 1569 is still using the be_substream. Not clear to me why you chose to protect the close() but not the hw_free()?

Reply:
actually you do protect the hw_free in the other case line 1946 now, so it's inconsistent.

Reply:
@plbossart Ran out of time, but I now realized some of the checks are superfluous as "be->dpcm[stream].state" can only be in limited states when BE substream is NULL. Need to go through all the additions. E.g. I think the case of L1946 is not needed.

Reply:
In dpcm_be_dai_stop(), be->dpcm[stream].users will be NULL as the BE is skipped in dai_startup(), so this check is not needed.